lib/chef/provider/package/chocolatey.rb

```diff
@@ -79,7 +79,7 @@ def install_package(names, versions)
           name_versions_to_install = desired_name_versions.select { |n, v| lowercase_names(names).include?(n) }
 
           name_nil_versions = name_versions_to_install.select { |n, v| v.nil? }
-          name_has_versions = name_versions_to_install.reject { |n, v| v.nil? }
+          name_has_versions = name_versions_to_install.compact
```

Comment:
I need to double check this one. This seems like it's not a compatible change

Reply:
I don't think this is going to do the same thing. You could replace the dual calls to select and reject with partition though:

```suggestion
          name_nil_versions, name_has_versions = name_versions_to_install.partition { |_, v| v.nil? }
```

Reply:
TIL compact actually does the right thing in a hash.

```
irb(main):001:0> {foo: "bar", baz: nil}
=> {:foo=>"bar", :baz=>nil}
irb(main):002:0> {foo: "bar", baz: nil}.compact
=> {:foo=>"bar"}
```

Reply:
wat

Reply:
TIL Hash has its own implementation of compact and doesn't just inherit from Enumerable or whatever I was expecting https://ruby-doc.org/core-2.7.2/Hash.html#method-i-compact

Reply:
I was all ready to fix this in RuboCop when I realized they're actually testing for it